src/ol/renderer/layerrenderer.js

```diff
@@ -191,7 +195,8 @@ ol.renderer.Layer.prototype.scheduleExpireCache =
              */
             function(tileSource, map, frameState) {
               var tileSourceKey = goog.getUid(tileSource).toString();
-              tileSource.expireCache(frameState.usedTiles[tileSourceKey]);
+              tileSource.expireCache(frameState.viewState.projection,
+                                     frameState.usedTiles[tileSourceKey]);
```

Comment:
You are right, but @petrsloup checks `canExpireCache()` first, and that only returns true when the cache count is over high water mark.